File: solutions/java/bob/1/src/main/java/Bob.java
import java.awt.*;

class Bob {

    private boolean isUppercase(String input) {
        if (input.isBlank())
            return false;

        input = input.chars().filter(Character::isLetter).collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append).toString();

        System.out.println(input);

        if (input.isEmpty())
            return false;

        for (char c : input.toCharArray())
            if (Character.isLowerCase(c))
                return false;

        return true;
    }

    private boolean isQuestion(String input) {
        input = input.trim();
        return input.isBlank() ? false : input.charAt(input.length() - 1) == '?';
    }

    String hey(String input) {
        boolean isUpper = isUppercase(input), isQuestion = isQuestion(input);

        if (isUpper && isQuestion){
            return "Calm down, I know what I'm doing!";
        } else if (isUpper) {
            return "Whoa, chill out!";
        } else if (isQuestion) {
            return "Sure.";
        } else if (input.isBlank()){
            return "Fine. Be that way!";
        }

        return "Whatever.";
    }
}

File: solutions/c/bob/1/bob.c
#include "bob.h"

#include <string.h>
#include <stdbool.h>
#include <ctype.h>

char *hey_bob(char *greeting)
{
    size_t greeting_len = strlen(greeting);

    char last_non_blank = '\0';
    bool all_whitespace = true;
    bool all_capital = false;

    // Check if there's any non-blank char
    // & save the last one to check if it's question mark
    for (size_t i = 0; i < greeting_len; i++)
    {
        char c = greeting[i];

        if (!isspace(c))
        {
            all_whitespace = false;
            last_non_blank = c;
        }
    }

    // Check if there's any alpha char that's not uppercase
    for (size_t i = 0; i < greeting_len; i++)
    {
        char c = greeting[i];

        if (isalpha(c))
            all_capital = true;

        if (islower(c))
        {
            all_capital = false;
            break;
        }
    }

    if (all_whitespace)
        return "Fine. Be that way!";

    if (last_non_blank == '?' && all_capital)
        return "Calm down, I know what I'm doing!";

    if (last_non_blank == '?')
        return "Sure.";

    if (all_capital)
        return "Whoa, chill out!";

    return "Whatever.";
}